Source/utils/SmallDataIOReader.cpp

```diff
@@ -0,0 +1,574 @@
+                      m_file_structure.num_data_columns.size() ==
+                          m_file_structure.num_blocks &&
+                      m_file_structure.num_columns.size() ==
+                          m_file_structure.num_blocks &&
+                      m_file_structure.block_starts.size() ==
+                          m_file_structure.num_blocks);
+    }
+    m_structure_defined = true;
+}
+
+// Set structure if known already (e.g. same as another file already
+// determined)
+void SmallDataIOReader::set_file_structure(
+    const SmallDataIOReader::file_structure_t &a_file_structure)
+{
+    m_file_structure = a_file_structure;
+    m_structure_defined = true;
+}
+
+// File structure getter
+const SmallDataIOReader::file_structure_t &
+SmallDataIOReader::get_file_structure(bool a_broadcast_to_all_ranks)
+{
+    if (a_broadcast_to_all_ranks)
+    {
+        // First convert std::vectors to Chombo Vectors to use Chombo broadcast
+        // functions
+        Vector<unsigned long long> block_starts_Vect =
+            m_file_structure.block_starts;
+        Vector<int> num_data_rows_Vect = m_file_structure.num_data_rows;
+        Vector<int> num_header_rows_Vect = m_file_structure.num_header_rows;
+        Vector<int> num_coords_columns_Vect =
+            m_file_structure.num_coords_columns;
+        Vector<int> num_data_columns_Vect = m_file_structure.num_data_columns;
+        Vector<int> num_columns_Vect = m_file_structure.num_columns;
+
+        broadcast(m_file_structure.num_blocks, 0);
+        broadcast(block_starts_Vect, 0);
+        broadcast(num_data_rows_Vect, 0);
+        broadcast(num_header_rows_Vect, 0);
+        broadcast(num_coords_columns_Vect, 0);
+        broadcast(m_file_structure.coords_width, 0);
+        broadcast(num_data_columns_Vect, 0);
+        broadcast(m_file_structure.data_width, 0);
+        broadcast(num_columns_Vect, 0);
+
+        m_file_structure.block_starts = block_starts_Vect.stdVector();
+        m_file_structure.num_data_rows = num_data_rows_Vect.stdVector();
+        m_file_structure.num_header_rows = num_header_rows_Vect.stdVector();
+        m_file_structure.num_coords_columns =
+            num_coords_columns_Vect.stdVector();
+        m_file_structure.num_data_columns = num_data_columns_Vect.stdVector();
+        m_file_structure.num_columns = num_columns_Vect.stdVector();
+    }
+    else
+    {
+        always_assert(procID() == 0);
+    }
+
+    return m_file_structure;
+}
+
+bool SmallDataIOReader::contains_data()
+{
+    if (!m_structure_defined)
+    {
+        determine_file_structure();
+    }
+    bool contains_data = false;
+
+    if (procID() == 0)
+    {
+        for (int iblock = 0; iblock < m_file_structure.num_blocks; ++iblock)
+        {
+            contains_data |= (m_file_structure.num_data_rows[iblock] > 0);
+        }
+    }
+#ifdef CH_MPI
+    // I think MPI_CXX_BOOL was added in MPI 3.0
+#if MPI_VERSION >= 3
```

Comment:
This code breaks the restart, so I propose we just remove it, and use the "else" option for all MPI versions, which works fine for me with a higher MPI version than 3.